lib/fabrik/Agent.js

```diff
@@ -19,8 +19,46 @@ class Agent extends HttpClient {
     this.settings = settings;
   }
 
-  get basePath() {
-    return `/v${this.settings.version}`;
+  basePath(host) {
+    const urls = [];
+    urls.push(formatUrl({
+      protocol: this.protocol,
+      hostname: host,
+      port: this.port,
+      pathname: path.posix.join(`/v1`, 'info')
+    }));
+    urls.push(formatUrl({
+      protocol: this.protocol,
+      hostname: host,
+      port: this.port,
+      pathname: 'info'
+    }));
+    return Promise
+      .any(_.map(urls, url => AGENT_CACHE[host] !== undefined ?
+        Promise.resolve(`/v${AGENT_CACHE[host].api_version}`) :
+        this
+        .request({
+          method: 'GET',
+          url: url
+        }, 200)
+        .then(res => {
+          if (_.isPlainObject(res.body)) {
+            return res.body;
+          }
+          throw new Error('Received invalid /info response', res);
+        })
+        .then(info => {
+          logger.debug(`${host} agent info - `, info);
+          let api_version = '1';
+          if (info && info.api_version &&
+            !Number.isInteger(info.api_version) &&
+            compareVersions(info.api_version, '1.3') < 0) {
+            info = _.set(info, 'api_version', `${api_version}`);
+          }
+          AGENT_CACHE[host] = info;
+          return `/v${info.api_version}`;
+        })
```

Comment:
Can you make this section a function and call here here ?  is part of a ternary operator and will help with readability.
```
 Promise.any(_.map(urls, url => AGENT_CACHE[host] !== undefined ?
        Promise.resolve(`/v${AGENT_CACHE[host].api_version}`) : getApiVersion(url)
      ));
```

Reply:
Agree with @muralisc

Reply:
Done